libraries/rush-lib/src/cli/actions/initSubspaceAction.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import type { CommandLineStringParameter } from '@rushstack/ts-command-line';
+import { assetsFolderPath } from '../../utilities/PathConstants';
+import type { RushCommandLineParser } from '../RushCommandLineParser';
+import { BaseRushAction } from './BaseRushAction';
+import type { ISubspacesConfigurationJson, SubspacesConfiguration } from '../../api/SubspacesConfiguration';
+import { FileSystem, JsonFile } from '@rushstack/node-core-library';
+
+const SUBSPACE_TEMPLATE_FOLDER_PATH: string = `${assetsFolderPath}/rush-init-subspace`;
+
+export class InitSubspaceAction extends BaseRushAction {
+  private readonly _subspaceName: CommandLineStringParameter;
+
+  public constructor(parser: RushCommandLineParser) {
+    super({
+      actionName: 'init-subspace',
+      summary: 'Create a new subspace.',
+      documentation:
+        'Use this command to create a new subspace with the default subspace configuration files.',
+      parser
+    });
+
+    this._subspaceName = this.defineStringParameter({
+      parameterLongName: '--name',
+      parameterShortName: '-n',
+      argumentName: 'SUBSPACE_NAME',
+      description: 'The name of the subspace that is being initialized.',
+      required: true
+    });
+  }
+
+  protected async runAsync(): Promise<void> {
+    if (!this.rushConfiguration.subspacesFeatureEnabled) {
+      throw new Error('The subspaces feature must be enabled to create a new subspace.');
+    }
```

Comment:
Should we even initialize this action if the subspaces feature isn't enabled?

Reply:
If subspaces was truly experimental, this might make sense.

However given that it's fully documented and fairly widely rolled out, I think it's fine to include it by default.

Also, I'd like to propose that `rush init-subspace` can automatically enable the feature and perform this file copying:

![image](https://github.com/microsoft/rushstack/assets/4673363/8509336a-5675-40fe-8989-f3d734516c06)

That could significantly simplify the instructions for setting up subspaces.

(@william2958 if we implement that idea, it can go in a separate PR.)